Prints the final scores once the game is over

Hunt-E opened the pull request:
Added a way to check if the game just got done with the last hole, and prints the final scores when it is.

abbiland commented:
Looks good to merge

nflanagan6 commented on src/csc439teamplatypus/cardgame/golfgame/CLIView.java:
```diff
@@ -256,6 +263,14 @@ protected void printPlayerScores() {
             System.out.println("Player " + playerRank[i] + ": " + currentScores[playerRank[i]]);
         System.out.println();
+
+        if (getNumberOfPlayedHoles() == getNumberOfHoles() - 1) {
```
@Hunt-E @roym5 @abbiland I might be misreading this, but I think this would display a winner if someone chose to view the current scores during the last hole before it was finished.

Hunt-E replied:
Yes it would, let me fix that.

nflanagan6 replied:
Thanks @Hunt-E

I have one more thing to double check, sorry I didn't see this earlier. If during the last hole (when numberOfPlayedHoles = numberOfHoles -1), a player takes the last card, but no player has all their cards up, the else if block in nextTurn() would end the game without flipping the face-down cards face up and adding them to the players' scores, right?

Hunt-E replied:
You are correct. This was unintended and was something that I overlooked. I am updating it now.

Hunt-E replied:
I worded that wrong because I skim read the question. The updateScores method adds the scores of all of the hands to the player scores without flipping them over. This is because the code is able to see the value of the card without ever flipping it over, and eliminates the extra lines it would take to flip them over when the deck is just going to be reset anyways. I am fixing the issue with not calling for updateScores to do that.

roym5 reviewed (commented):
good change

nflanagan6 commented on src/csc439teamplatypus/cardgame/golfgame/CLIView.java:
```diff
@@ -48,6 +48,7 @@ public void chooseCardToFlip() {
         while (!inputCompleted) {
 
             System.out.println("Which card would you like to flip over? You can only flip over a face-down card.");
+            printPlayerHand(getCurrentPlayerHand());
```
Thanks for adding this part in

Hunt-E commented:
What I changed:

updateScores inside of the else if statement in nextTurn so that it prints the finals cores once the game is over.

In printPlayerScores, I changed it to check if it's the end of a hole or not when setting current scores. I did this so that when it is called, it doesn't update the scores a second time with each player's hand after calling updateScores and then calling getPlayerScores when it is printing the end of each hole.

I added clears to the deck pile and discard pile inside of createCardDeck, because we call on it at the end of each hole, and it would just make a massive deck of the leftover cards if we didn't.

I added a getEndOfHoles method so that it just returns the scores that have last been through updateScores. This is used for printing the final scores of each hole so that all the cards don't have to be flipped up and waste time.

I updated incHole so that we can just call upon nextHole in CLIView and it just resets everything instead of calling on 3 different methods, essentially remaking the deck and dealing everyone's hands again as well as setting numberOfPlayedTurns back to 0 because of a new hole.

roym5 reviewed (commented):
your logic behind clearing the deck makes absolute sense and its good to clear it when you do.
what you do to incHole() is better for simplifying the logic to just one method instead of multiple.
good handling on printPlayerScores() as well for the turns score.

nflanagan6 commented on src/csc439teamplatypus/cardgame/golfgame/Controller.java:
```diff
@@ -47,6 +47,8 @@ protected boolean cardsRemaining() {
      */
     private void createCardDeck(ArrayList<GolfCard> deck, int numberOfDecks) {
+        deck.clear();
+        discard.clear();
```
Good catch clearing the discard

nflanagan6 commented:
Looks good @Hunt-E, thanks for doing this!
